src/memray/_memray/tracking_api.h

```diff
@@ -116,23 +122,22 @@ class NativeTrace
     }
     __attribute__((always_inline)) inline bool fill(size_t skip)
     {
-        d_data.resize(MAX_SIZE);
+        d_data.resize(128);
```

Comment:
You've changed this to call `d_data.resize()` on every call to `fill`, which defeats part of the optimization. We want to call this only in the constructor (if the vector is empty). Doing it here shrinks the vector if it was already larger than 128 items.